devel/scriptix/Portfile

```diff
@@ -1,23 +1,52 @@
-PortSystem 1.0
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem       1.0
 
 name             scriptix
 version          0.31
-revision         2
+revision         3
 categories       devel
 license          BSD
 maintainers      nomaintainer
 description      scripting system coded in C++
-long_description \
-	Scriptix is a scripting system coded in C++, designed solely \
-	for embedding in other C++ applications. Scriptix doesn't \
-	pretend to be a stand-alone language.
-homepage         http://www.awemud.net/
-platforms        darwin
+long_description Scriptix is a scripting system coded in C++, designed solely \
+                 for embedding in other C++ applications. Scriptix doesn't \
+                 pretend to be a stand-alone language.
+homepage         https://sourceforge.net/projects/awemud/
 master_sites     sourceforge:awemud
-checksums        md5 62cfa278d9580d17bedce9d67365cc93
+checksums        rmd160  6286fdffaf324d0f754340d921bb191f9dd8c82c \
+                 sha256  2b38ca0cad98f3794b9460ea1ee3f6f57ce3fd225a2854f32ad50eab6f403c47 \
+                 size    386220
+
 depends_lib      port:boehmgc
-patchfiles	 patch-Makefile.am patch-Makefile.in
+
+patchfiles       patch-Makefile.am.diff \
+                 patch-Makefile.in.diff \
+                 patch-missing-headers.diff
+
 post-patch {
-	reinplace s|shrext_cmds|shrext|g ${worksrcpath}/configure
+    reinplace s|shrext_cmds|shrext|g ${worksrcpath}/configure
 }
+
+platform darwin {
+    if {${os.major} > 10 && ${os.major} < 16} {
+        # Newer gcc and clang fail to build it.
+        # apple-gcc-4.2 builds through 10.11.
+        depends_build-append    port:apple-gcc42
+        configure.compiler      apple-gcc-4.2
+    }
+
+    if {${os.major} > 15} {
+        known_fail  yes
+        pre-fetch {
+            ui_error "${name} requires MacOS 10.11 or earlier"
+            return -code error "incompatible OS version"
+        }
+    }
+}
```

Comment:
While it makes sense to limit this to 10.11 or earlier, which compiler will be used for 10.4 thru 10.6?

Reply:
Xcode gcc. Should we specify it explicitly? I though it should be just picked.

Reply:
> Xcode gcc. Should we specify it explicitly? I thought it should be just picked.

Yep, just tested on 10.5, and builds fine. So we're good-to-go, no other changes necessary.